res/css/views/dialogs/security/_AccessSecretStorageDialog.scss

```diff
@@ -14,109 +14,139 @@ See the License for the specific language governing permissions and
+            align-items: center;
+
+            .mx_AccessSecretStorageDialog_recoveryKeyEntry_textInput {
+                flex-grow: 1;
+            }
+
+            .mx_AccessSecretStorageDialog_recoveryKeyEntry_entryControlSeparatorText {
+                margin: 16px;
+            }
+
+            .mx_AccessSecretStorageDialog_recoveryKeyEntry_fileInput {
+                display: none;
+            }
+        }
+
+        .mx_AccessSecretStorageDialog_recoveryKeyFeedback {
+            &::before {
+                content: "";
+                display: inline-block;
+                vertical-align: bottom;
+                width: 20px;
+                height: 20px;
+                mask-repeat: no-repeat;
+                mask-position: center;
+                mask-size: 20px;
+                margin-right: 5px;
+            }
+
+            &.mx_AccessSecretStorageDialog_recoveryKeyFeedback_valid {
+                color: $accent;
+
+                &::before {
+                    mask-image: url('$(res)/img/feather-customised/check.svg');
+                    background-color: $accent;
+                }
+            }
+
+            &.mx_AccessSecretStorageDialog_recoveryKeyFeedback_invalid {
+                color: $alert;
+
+                &::before {
+                    mask-image: url('$(res)/img/feather-customised/x.svg');
+                    background-color: $alert;
+                }
+            }
+        }
+
+        .mx_Dialog_buttons {
+            text-align: initial;
+            display: flex;
+            flex-flow: column;
+            gap: 14px;
+
+            .mx_Dialog_buttons_additive {
+                float: none;
+
+                .mx_AccessSecretStorageDialog_reset {
+                    position: relative;
+                    padding-left: 24px; // 16px icon + 8px padding
+
+                    &::before {
+                        content: "";
+                        display: inline-block;
+                        position: absolute;
+                        height: 16px;
+                        width: 16px;
+                        left: 0;
+                        top: 2px; // alignment
+                        background-image: url("$(res)/img/element-icons/warning-badge.svg");
+                        background-size: contain;
+                    }
+
+                    .mx_AccessSecretStorageDialog_reset_link {
+                        color: $alert;
+                    }
+                }
+            }
+
+            .mx_Dialog_buttons_items {
+                display: flex;
+                flex-wrap: wrap;
+                gap: 16px;
+                margin-left: auto;
+                padding-left: 24px; // See mx_AccessSecretStorageDialog_reset above
+
+                button {
+                    margin: 0 !important; // override rules on common.scss
+                }
+            }
+        }
     }
 }
-
-.mx_AccessSecretStorageDialog_titleWithIcon::before {
```

Comment:
Is this change in nesting strictly necessary? To me, it looks like these selectors are sufficiently specific. Unnecessary nesting of the styles makes them harder to read and harder to maintain.

Reply:
Selector's specificity itself does not ensure correct styling, as you have seen on https://github.com/matrix-org/matrix-react-sdk/pull/8089. Nesting should save you from unnoticed visual regressions otherwise. It should also clarify the code structure between scss and tsx files. I strongly believe in the value of nesting, but I am afraid of it that it is just an amateur view of an outsider...

If you do not accept nesting, I am fine to drop those commits.

Reply:
Still, I would like to clarify that it is nesting of `.mx_AccessSecretStorageDialog_reset` inside `.mx_Dialog_buttons_additive` that fixes the bug of the link color. Currently `color: $alert;` of `mx_AccessSecretStorageDialog_reset_link` is overridden by `.mx_AccessibleButton_kind_link_inline` on Firefox. In order to prevent similar regressions from happening, nesting should work.